### ISSUE TITLE
Use `INSERT ... ON CONFLICT DO NOTHING` instead of unexpected SELECT + INSERT on calling `upsert` with empty `update`

### ISSUE THREAD

**Issue by rgmvisser** [prisma/prisma#20229]
### Bug description

TLDR:
When calling upsert with an empty update object prisma does a SELECT + INSERT instead of a INSERT ... ON CONFLICT DO NOTHING. 

More context here: https://github.com/prisma/docs/pull/5048#issuecomment-1636050244

### How to reproduce

Do something like:
```
prisma.object.upsert({
      where: {
          hash
      },
      create: {
        hash,
        tenantId: tenant.id,
      },
      update: {},
    });
````

It will produce a select and than an insert. However, when dealing with potential race conditions (calling this twice at the same time for instance), the order might be: SELECT 1-> no result, SELECT  2-> no result,  INSERT 1 -> new row, INSERT 2 -> Error of duplicated unique field "hash"

If it would be a native UPDATE ON CONFLICT, this would be handled in the case of a race condition by the DB. 

At the moment I solve this by calling the function like this:
```
prisma.object.upsert({
      where: {
          hash
      },
      create: {
        hash,
        tenantId: tenant.id,
      },
      update: {
            hash
      },
    });
````

This forces Prisma to do a native upsert (even though the hash never changes/updates). 

### Expected behavior

When calling upsert with an empty update object, it should still call INSERT ... ON CONFLICT DO NOTHING, so that a race condition is still handled.

### Prisma Version

```
4.9
```

**Comment by janpio** [comment]
For context, the queries that are generated by a `upsert()` with an empty `update` object:

Data already exists:
```
Query: BEGIN
Params: []
Query: SELECT "public"."User"."id" FROM "public"."User" WHERE ("public"."User"."email" = $1 AND 1=1) OFFSET $2
Params: ["ruheni@prisma.io",0]
Query: SELECT "public"."User"."id" FROM "public"."User" WHERE ("public"."User"."email" = $1 AND 1=1)
Params: ["ruheni@prisma.io"]
Query: SELECT "public"."User"."id", "public"."User"."createdAt", "public"."User"."updatedAt", "public"."User"."email", "public"."User"."name", "public"."User"."age", "public"."User"."country", "public"."User"."role" FROM "public"."User" WHERE "public"."User"."id" = $1 LIMIT $2 OFFSET $3
Params: [2,1,0]
Query: COMMIT
Params: []
```

Data does not exist yet:
```
Query: BEGIN
Params: []
Query: SELECT "public"."User"."id" FROM "public"."User" WHERE ("public"."User"."email" = $1 AND 1=1) OFFSET $2
Params: ["janpio@prisma.io",0]
Query: INSERT INTO "public"."User" ("createdAt","updatedAt","email","name","age","country","role") VALUES ($1,$2,$3,$4,$5,$6,$7) RETURNING "public"."User"."id"
Params: ["2022-12-21 12:52:42.087 UTC","2022-12-21 12:52:42.087 UTC","janpio@prisma.io","Jan",23,"Germany","USER"]
Query: SELECT "public"."User"."id", "public"."User"."createdAt", "public"."User"."updatedAt", "public"."User"."email", "public"."User"."name", "public"."User"."age", "public"."User"."country", "public"."User"."role" FROM "public"."User" WHERE "public"."User"."id" = $1 LIMIT $2 OFFSET $3
Params: [10,1,0]
Query: COMMIT
```

**Comment by rgmvisser** [comment]
Interesting edge case, thanks for the explanation @laplab

**Comment by karlkeefer** [comment]
This behavior was very surprising to me. I also don't fully grok @laplab 's explanation above for why it should error - the linked postgres docs make my head spin.

What about a condition where we know the where clause and create clause _do_ have the same value. My use case looks like this:

```
events.upsert({
  where: {
    uuid: "abcd"
  },
  update: {},
  create: {
    uuid: "abcd",
    title: "example"
  },
}),
```
 
The docs outline [conditions where "database upsert" will be used](https://www.prisma.io/docs/orm/reference/prisma-client-reference#database-upsert-query-criteria). The above query satisfies those constraints but still doesn't use `ON CONFLICT`...

It seems like either this query should use ON CONFLICT even if `update: {}`, or I should make a PR against the docs adding a condition that the update clause must be non-empty 🤔

**Comment by RedForest347** [comment]
Can you please update docs and add the "update is not empty" condition to the [Database upsert query criteria](https://www.prisma.io/docs/orm/reference/prisma-client-reference#database-upsert-query-criteria) paragraph?
